llvm/lib/Target/X86/X86InsertWait.cpp

```diff
@@ -103,8 +103,9 @@ bool WaitInsert::runOnMachineFunction(MachineFunction &MF) {
 
   for (MachineBasicBlock &MBB : MF) {
     for (MachineBasicBlock::iterator MI = MBB.begin(); MI != MBB.end(); ++MI) {
-      // Jump non X87 instruction.
-      if (!X86::isX87Instruction(*MI))
+
+      // Jump non X87 instruction and Inline asm.
+      if (!X86::isX87Instruction(*MI) || MI->isInlineAsm())
```

Comment:
Put the check in `isX87Instruction`?

Reply:
Updated, I did -ammend and force pushed. please do let me know if this is something unacceptable thanks.

Reply:
Oops my bad  I accidentally force pushed incomplete changes, will update today sorry for any confusions.